pkgs/applications/networking/browsers/yandex-browser/default.nix

```diff
@@ -147,29 +167,49 @@ in stdenv.mkDerivation rec {
     substituteInPlace $out/share/applications/${pname}.desktop --replace /usr/ $out/
     substituteInPlace $out/share/menu/yandex-browser${app}.menu --replace /opt/ $out/opt/
     substituteInPlace $out/share/gnome-control-center/default-apps/yandex-browser${app}.xml --replace /opt/ $out/opt/
-    ln -sf ${vivaldi-ffmpeg-codecs}/lib/libffmpeg.so $out/opt/yandex/browser${app}/libffmpeg.so
+    ln -sf ${yandex-ffmpeg-codecs}/lib/libffmpeg.so $out/opt/yandex/browser${app}/libffmpeg.so
     ln -sf $out/opt/yandex/browser${app}/yandex-browser${app} $out/bin/${pname}
   '';
 
-  runtimeDependencies = map lib.getLib [
-    libpulseaudio
-    curl
-    systemd
-    vivaldi-ffmpeg-codecs
-  ] ++ buildInputs;
+  postFixup = ''
+    patchelf --add-needed "libGL.so.1" "$out/opt/yandex/browser${app}/yandex_browser"
+    wrapProgram "$out/opt/yandex/browser${app}/yandex_browser" \
+                 --add-flags ${lib.escapeShellArg commandLineArgs} \
+                 --add-flags "--disable-gpu-compositing" \
+                 --add-flags "--in-process-gpu" \
+                 --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}}" \
+                 --set-default FONTCONFIG_FILE "${fontconfig.out}/etc/fonts/fonts.conf" \
+                 --set-default FONTCONFIG_PATH "${fontconfig.out}/etc/fonts" \
+                 --set-default LIBGL_DRIVERS_PATH "${mesa.drivers}/lib:${mesa.drivers}/lib/dri" \
+                 --set-default LIBVA_DRIVERS_PATH "${mesa.drivers}/lib:${mesa.drivers}/lib/dri" \
```

Comment:
Shouldn't this be using `/run/opengl-driver`?

`addDriverRunpath.driverLink` exposes this path

Reply:
will refer for this: --prefix XDG_DATA_DIRS  : "${addDriverRunpath.driverLink}/share" ?